File: d2022_05_07_Problema_00456/modelo5.go
func find132pattern(nums []int) bool {
    
    min := nums [0] // Sempre haverá
    
    for j := 1; j < len (nums) - 1; j++ {
        if min < nums [j] {
            max := nums [j]
            for k := j + 1; k < len (nums); k++ {
                if nums [k] < max {
                    if nums [k] > min {
                        return true
                    }
                } else {
                    max = nums [k]
                }
            }
        } else {
            min = nums [j]
        }
    }
    
    return false
}

File: d2025_07_25_Problema_03487/modelo5.rs
use std::collections::HashSet;
impl Solution {
    pub fn max_sum(nums: Vec<i32>) -> i32 {
        if nums.iter ().any (|v| *v >= 0) {
            return HashSet::<i32>::from_iter::<Vec<i32>> (nums.into_iter ().filter (|v| *v > 0).collect::<_> ()).into_iter ().sum ();
        }
        // else
        nums.into_iter ().filter (|v| *v < 0).max ().unwrap () // Se chegar aqui é porque haverá pelo menos um
    }
}
